core/src/main/java/org/apache/iceberg/util/Tasks.java

```diff
@@ -273,6 +269,16 @@ public boolean run(Task<I, RuntimeException> task) {
       return !threw;
     }
 
+    private void tryRunOnFailure(I item, Exception failure) {
+      try {
+        onFailure.run(item, failure);
+      } catch (Exception failException) {
+        failure.addSuppressed(failException);
+        LOG.error("Failed to clean up on failure", failure);
```

Comment:
I know that this was what we originally had, but the logged error seems incorrect. Don't we want to log `failException`? The `failure` should be propagated up anyways.

Reply:
I think you're right. It should log `failException`. At least we're adding it as a suppressed exception so it is still available.